nuxeo-vision-core/src/main/java/org/nuxeo/vision/core/service/VisionImpl.java

```diff
@@ -87,136 +69,86 @@ public void deactivate(ComponentContext context) {
     }
 
     /**
-     * Application started notification. Called after the application started.
-     * You can do here any initialization that requires a working application
-     * (all resolved bundles and components are active at that moment)
+     * Application started notification. Called after the application started. You can do here any initialization that
+     * requires a working application (all resolved bundles and components are active at that moment)
      *
-     * @param context the component context. Use it to get the current bundle
-     *            context
+     * @param context the component context. Use it to get the current bundle context
      * @throws Exception
      */
     @Override
     public void applicationStarted(ComponentContext context) {
     }
 
     @Override
-    public void registerContribution(Object contribution,
-            String extensionPoint, ComponentInstance contributor) {
+    public void registerContribution(Object contribution, String extensionPoint, ComponentInstance contributor) {
         if (CONFIG_EXT_POINT.equals(extensionPoint)) {
             config = (VisionDescriptor) contribution;
-        } else if (GOOGLE_EXT_POINT.equals(extensionPoint)) {
-            googleConfig = (GoogleVisionDescriptor) contribution;
+        } else if (PROVIDER_EXT_POINT.equals(extensionPoint)) {
+            VisionProviderDescriptor desc = (VisionProviderDescriptor) contribution;
+            try {
+                VisionProvider provider = (VisionProvider) desc.getClassName().getConstructor(Map.class).newInstance(
+                        desc.getParameters());
+                providers.put(desc.getProviderName(), provider);
+            } catch (InstantiationException | IllegalAccessException | NoSuchMethodException
+                    | InvocationTargetException e) {
```

Comment:
Use `ReflectiveOperationException` instead of the several individual subclasses of it to shorten this.